src/Azure.Deployments.Extensibility.Providers.Graph/GraphProvider.cs

```diff
@@ -0,0 +1,399 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Deployments.Extensibility.Core;
+using Azure.Deployments.Extensibility.Core.Json;
+using Azure.Deployments.Extensibility.Core.Exceptions;
+using Azure.Deployments.Extensibility.Core.Validators;
+using Json.Pointer;
+using System.Net;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+namespace Azure.Deployments.Extensibility.Providers.Graph
+{
+    public class GraphProvider : IExtensibilityProvider
+    {
+        public static readonly string ProviderName = "Graph";
+
+        private static readonly string InternalError = "InternalError";
+        private static readonly string DisplayName = "displayName";
+        private static readonly string GraphToken = "graphToken";
+        private static readonly string Name = "name"; // name is required in Bicep template for Graph extensible types
+        private static readonly string Id = "id";
+        private static readonly string Value = "value";
+        private static readonly IReadOnlyDictionary<string, string> AlternateKeyByType = new Dictionary<string, string>()
+        {
+            ["users"] = "userPrincipalName",
+        };
+
+        // This is a temporary solution for POC to recognize operations on entity collections
+        // TODO: Find a more consistent way and type generic way to handle the cases
+        private static readonly IReadOnlySet<string> NavigationPropertyByType = new HashSet<string>()
+        {
+            "Microsoft.Graph/groups/members@2022-06-15-preview",
+            "Microsoft.Graph/servicePrincipals/appRoleAssignments@2022-06-15-preview",
+        };
+
+        private readonly GraphHttpClient GraphHttpClient;
+
+        public GraphProvider()
+        {
+            this.GraphHttpClient = new GraphHttpClient();
+        }
+
+        public GraphProvider(GraphHttpClient client)
+        {
+            this.GraphHttpClient = client;
+        }
+
+        public Task<ExtensibilityOperationResponse> DeleteAsync(ExtensibilityOperationRequest request, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<ExtensibilityOperationResponse> GetAsync(ExtensibilityOperationRequest request, CancellationToken cancellationToken)
+        {
+            return Execute(ProcessGetAsync)(request, cancellationToken);
+        }
+
+        public Task<ExtensibilityOperationResponse> PreviewSaveAsync(ExtensibilityOperationRequest request, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
```

Comment:
`PreviewSave` must be implemented - the deployment engine will always call `previewSave` before `save`. For the demo we can just return a response with `request.Resource`.